lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -158,6 +158,26 @@ function dosomething_northstar_get_northstar_user($drupal_id) {
 }
 
 /**
+ * Get a Northstar user for a given index & type. Useful for non Drupal ID lookups.
+ * @param  string $index_type The type of index to query for (Email, Facebook, Mobile, _id)
+ * @param  string $index The index to query for
+ * @return object|null
+ */
+function dosomething_northstar_get_user_by_custom_index($index_type, $index) {
```

Comment:
ah, sorry I didn't catch this before, but we already have a `dosomething_northstar_get_northstar_user` which grabs by `drupal_id` but I'd love to take the approach of having *less* methods and refactor that one a bit so it can take a `$type` and a `$key` (which is I think what they are called in NS) instead of creating a new method to do essentially the same thing just for a different type of `get`